### PR TITLE
Implement programmatic mode

### LINKED ISSUE #61: Feature: Programmatic usage
Story: `As a developer I want to import this package inside my project so that build tools on top of it.`

The goal is to make possible to import the tool to your project, so I can use all CLI features inside a project you have.

PS: this issue will not be focussed until #59 is closed.

### PR DESCRIPTION
**Pull request by BrOrlandi**
Resolves #61

### CODE REVIEW

File: src/cli/execution.js [hunk 1]
```diff
@@ -15,7 +32,10 @@ const execution = () => {
     return cli.showHelp(1)
   }
 
-  return scanFromFile(cli.input[0], cli.flags)
```
Comment (victorperin) [inline src/cli/execution.js]: I don't think moving scanFromFile responsability to here is the best way to do it. What about a flag or a third argument to pass what the output should be?

File: src/infrastructure/qrcode-reader.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ const readQR = imageBitmap =>
     const qr = new QrCode()
 
     qr.callback = (error, value) => {
-      if (error) return reject(error)
+      if (error) return reject(new Error(error))
```
Comment (victorperin) [inline src/infrastructure/qrcode-reader.js]: This isn't already an error that comes from qrcode-reader package?

File: src/cli/execution.js (hunk 1 above)
Comment (victorperin) [reply src/cli/execution.js]: This is bad for 2 reasons:
- we can't reuse the options on cli.
- execution files becames a lot more complex.

File: src/infrastructure/qrcode-reader.js (hunk 2 above)
Comment (BrOrlandi) [reply src/infrastructure/qrcode-reader.js]: No, it's a String.

Comment (victorperin) [reply src/infrastructure/qrcode-reader.js]: I just discovered that, at least, 2 integration tests are passing without assertions checking. This is a problem and a bug.

**Comment by victorperin** [comment]
@BrOrlandi I updated and fixed all those structure problems. Can you implement this again?

**Comment by victorperin** [comment]
@BrOrlandi we are now prepared to this implementation. Also typescript helps a lot with self documentation of code.

**Pull request closed without merging**